### ISSUE TITLE
[Postgres] geometry/geography subtypes in function arguments fail to parse

### ISSUE THREAD

**Issue by dmfay** [sqlfluff/sqlfluff#6351]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
create or replace function test (
  _pt geography(point)
)
returns numeric
language sql
begin atomic
  select st_x(_pt::geometry);
end;
```

### Expected Behaviour

successful parse

### Observed Behaviour

`PRS: Line 1, Position 1: Found unparsable section: 'create or replace function test....`

Removing the `(point)` on line 2 fixes the issue.

### How to reproduce

Copy "what happened" into editor.

### Dialect

Postgres

### Version

3.2.3, installed using Mason.nvim

### Configuration

```toml
[sqlfluff]
dialect = postgres
max_line_length = 92

# L003: can't seem to get indentation to like "two spaces inside a begin/end block"
# L031: you can take my aliases from my cold, dead hands
# L032: ditto USING
# L042: and joined subqueries
# L061: and <>
# LT01: don't require space between type and square brackets for arrays
# LT08: don't require blank lines around ctes
exclude_rules =
  L003,
  L031,
  L032,
  L042,
  L061,
  LT01,
  LT08

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy=None
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by R3gardless** [comment]
@dmfay 
I'm not very familiar with PostGIS syntax, but I believe this bug is related to the code at this link: (https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_postgres.py#L1428).

**Comment by hectorj-thetreep** [comment]
workaround : `geography(point, 4326)` seems to be accepted
